src/Products/ZCatalog/query.py

```diff
@@ -89,8 +89,8 @@ def __init__(self, request, iid, options=(), operators=('or', 'and'),
             for field in request.keys():
                 if field.startswith(iid + '_'):
                     iid_tmp, op = field.split('_')
```

Comment:
```suggestion
                    tmp, op = field.split(iid + '_')
```
This avoids ambiguity, if the option contains an underscore too.